wordpress-mobile/WordPress-Android
# Stats revamp v2 total comments detail screen

This PR implements **Total Comments** detail screen with **Total Comments**, **Top Commentators** and **Posts Comments** cards

- [x] To do Update [FluxC part](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2386) before merge

Fixes #16508  


<img src="https://user-images.githubusercontent.com/990349/167792304-aa64e1a7-25d3-4458-be27-b0d6c50d7fe6.png" width=320>


To test:

Setup:

-  Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
-  Select Debug Settings
-  Find StatsRevampV2FeatureConfig under Features in development enable it and restart app

Test 1:

- Launch app
- Go to stats either using quick links or menu
- Ensure that it opens in Insights tab otherwise switch to Insights tab
- Notice a new Top Comments card is shown in Insights tab (scroll down if necessary)
- If the Total Comments cards is not present, go to the stats management (either through ⚙️ menu at the top right hand corner or Add new stats card at the bottom) and add the Total Comments card and save, then return to Insights.
- Tap on VIEW_MORE on the top right hand corner on the Total Comments card
- Ensure it opens the detail screen as shown above with **Total Comments**, **Top Commentators** and **Posts Comments** cards (scroll down if necessary)


NOTE:

- This PR is only for navigation and detail screen with relevant existing cards
- Cards themselves will be updated in their own PRs
- Date selector is hidden as per discussions. Will add in a separate PR if required


## Regression Notes
1. Potential unintended areas of impact
None, as it is a new screen

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit testing and Manual testing

3. What automated tests I added (or what prevented me from doing so)
existing unit tests

PR submission checklist:
